Formula/ruby@1.8.7.rb

```diff
@@ -0,0 +1,205 @@
+class RubyAT187 < Formula
+  desc "Powerful, clean, object-oriented scripting language"
+  homepage "https://www.ruby-lang.org/"
+  url "https://cache.ruby-lang.org/pub/ruby/1.8/ruby-1.8.7-p374.tar.bz2"
+  sha256 "b4e34703137f7bfb8761c4ea474f7438d6ccf440b3d35f39cc5e4d4e239c07e3"
+  revision 2
+
+  bottle do
+    sha256 "bc89a74b22eec03454e42a37ce12df2fd31765e3ed9f87673fb3d073f0d3db05" => :sierra
+    sha256 "85ffdf739e1f4f74979e59616933e57a3a384bfc2214277b34120295ab0a695d" => :el_capitan
+    sha256 "f40e11d81cdae7051600b72ea5511b9995420618a13b65be439f6978e634d8c5" => :yosemite
+  end
+
+  option :universal
+  option "without-suffix", "Do not suffix commands with '187' (will become keg-only)"
+  option "with-doc", "Install documentation"
+  option "with-tcltk", "Install with Tcl/Tk support"
+
+  provided_by_osx = [:mountain_lion, :lion, :snow_leopard].include? MacOS.version.to_sym
+  if provided_by_osx && (build.without? "suffix")
```

Comment:
I have marked a given ruby version as keg-only only if the same version is provided by Apple. Leopard (10.5) has ruby 1.8.6, Mavericks (10.9) has ruby 2.0.0.

I wasn't aware of the provided_pre shortcut. Note the comparison is strict so should be `:provided_pre_mavericks`.    That could be done if we drop Leopard entirely which comes with ruby 1.8.6.

For reference from opensource.apple.com:

| Ver | Name | Ruby version
|--|--|--
| 10.12 | sierra | 2.0.0-p648
| 10.11 | el_capitan | 2.0.0-p648
| 10.10 | yosemite | 2.0.0-p481
| 10.9 | mavericks | 2.0.0-p481
| 10.8 | mountain_lion | 1.8.7-p357
| 10.7 | lion | 1.8.7-p357
| 10.6 | snow_leopard | 1.8.7-p174
| 10.5 | leopard | 1.8.6-p287